src/Storages/MergeTree/MergeTreeDataPartDistinctPartitionExpressionCloner.cpp

```diff
@@ -37,7 +37,12 @@ void MergeTreeDataPartDistinctPartitionExpressionCloner::updateMinMaxFiles(
 {
     deleteMinMaxFiles(storage_builder);
 
-    [[maybe_unused]] auto written_files = dst_part->minmax_idx->store(*merge_tree_data, storage_builder, dst_part->checksums);
+    auto written_files = dst_part->minmax_idx->store(*merge_tree_data, storage_builder, dst_part->checksums);
+
+    for (auto & file : written_files)
+    {
+        file->finalize();
```

Comment:
Let's also put a TODO, that `sync` is missing here (but it is not urgent, so you can bulk this change with others)

Reply:
Added the parameter to control sync and hard-coded it to false on caller side